main.tf

```diff
@@ -39,10 +39,20 @@ resource "aws_route53_zone" "zone" {
 locals {
   records_expanded = [
     for record in var.records : merge({
-      record_id = join("-", compact([lower(record.type), lower(record.name)]))
-      ttl       = null
-      records   = null
-      alias     = null
+      record_id = join("-", compact([lower(record.type), lower(record.name),
+        try(
+          "${lower(record.set_identifier)}-${lower(record.weight)}",                                    # weighted routing policy
```

Comment:
doesn't identifier so the job as this is something you can set in a unique way... adding other values seems too much as any change would create a new and delete another resource... makes plans really weird.. as it is not easy to spot recreation then..